src/lightning_lite/plugins/collectives/torch_.py

```diff
@@ -100,6 +100,8 @@ def scatter_object_list(
         return scatter_object_output_list
 
     def barrier(self, device_ids: Optional[List[int]] = None) -> None:
+        if dist._rank_not_in_group(self.group):
+            return
```

Comment:
Without this, we would need to check the global rank before calling `barrier` as it was done in https://github.com/Lightning-AI/lightning/pull/15195/commits/baa6f91fa8758e23ea4f3e70d12917d0028d5de7 to avoid the warning:
```python
UserWarning: Running barrier on global rank 0 which does not belong to the given group.
```

Reply:
I think this is a general issue for any compute if collectives are called on the wrong rank and adding this check can't hurt at all, right? It would make it a lot easier that user's don't have to pay attention on which rank to call which collectives but to always call them on any rank and it doesn't do wrong stuff (or anything at all) unless you're on a correct rank.

Reply:
I don't think it hurts for `barrier` specifically. But perhaps the warning is useful to uncover issues in the code for the other collectives.

Note that if we dont' have this, it still works correctly, it just raises the warning.

Reply:
here it does, maybe somewhere else it doesn't... 

But I think that this would be more consistent in general anyways. Meaning that if we create something we also destroy it if no longer needed